azext_edge/edge/providers/check/mq.py

```diff
@@ -975,7 +975,28 @@ def _evaluate_listener_service(
     namespace: str,
     detail_level: int = ResourceOutputDetailLevel.summary.value,
 ) -> None:
-    listener_spec_service_name: str = listener_spec["serviceName"]
+    listener_spec_service_name: str = listener_spec.get("serviceName")
+
+    # skip following check if service name is not specified
+    if not listener_spec_service_name:
+        listener_service_eval_status = CheckTaskStatus.error.value
+        check_manager.add_display(
+            target_name=target_listeners,
+            namespace=namespace,
+            display=Padding(
+                f"\n{colorize_string(color='red', value='Invalid')} service name {{{colorize_string(color='red', value=listener_spec.get('name'))}}} for listener.",
+                (0, 0, 0, 12),
+            ),
+        )
+        check_manager.add_target_eval(
+            target_name=target_listeners,
+            namespace=namespace,
+            status=listener_service_eval_status,
+            value={"listener_service": "Unable to fetch service name."},
+            resource_name=f"service/{listener_spec_service_name}",
```

Comment:
wondering if this error should be targeted for the listener resource name rather than the service name? Since there's no `listener_spec_service_name`, this will always be `service/None` - and I believe the condition for `spec.serviceName` belongs to the listener itself. What about a value of {`spec.serviceName`: `None`} and a `resource_name` of the listener?

Reply:
like this? 😊
<img width="143" alt="image" src="https://github.com/user-attachments/assets/ad8cff72-9fa6-4e64-b13b-04500c309df3" />